router/health_check.go

```diff
@@ -0,0 +1,123 @@
+package router
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+
+	"github.com/buildwithgrove/path/relayer/morse"
+	"github.com/buildwithgrove/path/relayer/shannon"
+	"github.com/pokt-network/poktroll/pkg/polylog"
+)
+
+// HealthCheck is an interface that must be implemented
+// by components that need to report their health status
+type HealthCheck interface {
+	// Name returns the name of the component being checked.
+	Name() string
+	// IsAlive returns true if the component is healthy, otherwise false.
+	IsAlive() bool
+}
+
+// All components that implement the HealthCheck interface must be registered here.
+var _ HealthCheck = &morse.Protocol{}
+var _ HealthCheck = &shannon.Protocol{}
+
+// One of `ready` or `not_ready`.
```

Comment:
Make comments ever green. Developers usually forget to update comments if the implementation changes and the comment is not immediately visible to them.

```suggestion
// The status of the health check component.
```

Reply:
Updated.